test/integration/puppetlabs/general_puppet/general_puppet_int_test.clj

```diff
@@ -273,4 +279,18 @@
      app {:jruby-puppet
           {:max-active-instances num-jrubies}}
      (let [catalog (testutils/get-catalog)]
-       (is (= "production" (get catalog "version")))))))
\ No newline at end of file
+       (is (= "production" (get catalog "version")))))))
+
+(deftest ^:integration custom-trusted-oid-mapping-test
+  (testing "custom_trusted_oid_mapping works properly"
+    (testutils/write-site-pp-file
+      "notify { 'trusted_hash':\n\tmessage => \"${trusted['extensions']}\",\n}")
+    (fs/copy custom-oid-mapping (fs/file testutils/conf-dir "custom_trusted_oid_mapping.yaml"))
+    (fs/copy csr-attributes (fs/file testutils/conf-dir "csr_attributes.yaml"))
+    (fs/delete-dir (fs/file testutils/conf-dir "ssl"))
+    (bootstrap/with-puppetserver-running
+      app
+      {}
+      (let [catalog (testutils/get-catalog)]
+        (is (= "{sf => Burning Finger, short => 22}"
+               (get-in (nth (get catalog "resources") 3) ["parameters" "message"])))))))
```

Comment:
Is the order in which this resource appears guaranteed?  Might be better to look for the first resource matching a title of `trusted_hash` and validating the content within it?